### PR TITLE
Add possibility to delete branch Model Condition

### PR DESCRIPTION
**Pull request by j0nathan33**

### CODE REVIEW

**Comment by cdancy** [comment]
Rebase off of master to resolve conflicts?

File: src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java [hunk 1]
```diff
@@ -146,18 +147,66 @@ public void testGetBranchModelConfiguration() {
...
+        types.add(Type.create(Type.TypeId.HOTFIX, null, "hot/", true));
+        types.add(Type.create(Type.TypeId.RELEASE, null, "rel/", true));
+        types.add(Type.create(Type.TypeId.FEATURE, null, "fea/", true));
+        CreateBranchModelConfiguration configuration = CreateBranchModelConfiguration.create(branchModelConfiguration.development(), null, types);
+
+
+        BranchModelConfiguration branchModelConfiguration = api().updateModelConfiguration(projectKey, repoKey, configuration);
+        assertThat(branchModelConfiguration).isNotNull();
+        assertThat(branchModelConfiguration.errors().isEmpty()).isTrue();
+        assertThat(branchModelConfiguration.development().refId()).isNotNull();
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java]: Removed this check in master as it was null for me on my version of Bitbucket.

File: src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java [hunk 2]
```diff
@@ -146,18 +147,66 @@ public void testGetBranchModelConfiguration() {
...
         assertThat(configuration.types()).isEmpty();
+        assertThat(configuration.production()).isNull();
+        assertThat(configuration.development()).isNull();
+        assertThat(configuration.errors()).isNotEmpty();
     }
 
     private void checkDefaultBranchConfiguration() {
         assertThat(branchModelConfiguration).isNotNull();
         assertThat(branchModelConfiguration.errors().isEmpty()).isTrue();
+        assertThat(branchModelConfiguration.development().refId()).isNotNull();
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java]: Removed this check as well as it was null for me.

File: src/main/java/com/cdancy/bitbucket/rest/features/BranchApi.java [hunk 3]
```diff
@@ -126,6 +127,23 @@ BranchModel model(@PathParam("project") String project,
...
+    BranchModelConfiguration updateModelConfiguration(@PathParam("project") String project,
+                                                      @PathParam("repo") String repo,
+                                                      @BinderParam(BindToJsonPayload.class) CreateBranchModelConfiguration config);
+
+    @Named("branch:delete-model-configuration")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/branch-utils/{jclouds.api-version}/projects/{project}/repos/{repo}/branchmodel/configuration")
+    @Fallback(BitbucketFallbacks.FalseOnError.class)
+    @DELETE
+    boolean deleteModelConfiguration(@PathParam("project") String project,
```
Comment (cdancy) [inline src/main/java/com/cdancy/bitbucket/rest/features/BranchApi.java]: @Documentation annotation for this? Searched online and couldn't find anything

Comment (j0nathan33) [inline src/main/java/com/cdancy/bitbucket/rest/features/BranchApi.java]: Yup, this endpoint isn't in documentation.

Source : https://jira.atlassian.com/browse/BSERV-5411

Comment (cdancy) [inline src/main/java/com/cdancy/bitbucket/rest/features/BranchApi.java]: @j0nathan33 can you add the Documentation annotation and use that jira ISSUE?

File: .gitattributes [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # Handle line endings automatically for files detected as text
 # and leave all files detected as binary untouched.
-* text=auto
+* text=lf
```
Comment (cdancy) [inline .gitattributes]: Can we back this out and open a separate PR for it if you truly want to change it?

File: src/main/java/com/cdancy/bitbucket/rest/features/BranchApi.java (hunk 3 above)
Comment (j0nathan33) [inline src/main/java/com/cdancy/bitbucket/rest/features/BranchApi.java]: ok, I will do it.

File: .gitattributes (hunk 4 above)
Comment (j0nathan33) [inline .gitattributes]: Ok, I will do it

**Comment by cdancy** [comment]
@j0nathan33 merged this PR with various fixes to account for proper integration test support now working OOTB. For future PR's, and assuming you have a bitbucket instance, simply running `./gradlew integTest` will now work for all live tests (also assumes you've set the necessary properties).

**Pull request merged**